source/features/ci-link.tsx

```diff
@@ -5,31 +5,51 @@ import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 import fetchDom from '../helpers/fetch-dom';
-import {buildRepoURL} from '../github-helpers';
+import {buildRepoURL, getConversationNumber} from '../github-helpers';
 
 // Look for the CI icon in the latest 2 days of commits #2990
-const getIcon = onetime(async () => fetchDom(
+const getRepoIcon = onetime(async () => fetchDom(
 	buildRepoURL('commits'), [
 		'.commit-group:nth-of-type(-n+2) .commit-build-statuses', // Pre "Repository refresh" layout
 		'.TimelineItem--condensed:nth-of-type(-n+2) .commit-build-statuses'
 	].join()
 ));
 
-async function init(): Promise<false | void> {
-	const icon = await getIcon() as HTMLElement | undefined;
+const getPRIcon = onetime(async () => fetchDom(
+	buildRepoURL('pull', getConversationNumber()!, 'commits'), '.js-commits-list-item:last-of-type .commit-build-statuses'
+));
+
+async function initRepo(): Promise<false | void> {
+	const icon = await getRepoIcon() as HTMLElement | undefined;
```

Comment:
Let's try to move these types to the `get*Icon` functions so we avoid assertions